constantine/build.gradle

```diff
@@ -0,0 +1,166 @@
+plugins {
+    id 'java-library'
+    id 'maven-publish'
+    id 'com.jfrog.artifactory' version '4.20.0'
+}
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    implementation 'net.java.dev.jna:jna:5.12.1'
+    testImplementation 'junit:junit:4.13.2'
+}
+
+java {
+    toolchain {
+        languageVersion = JavaLanguageVersion.of(21)
+    }
+}
+
+//task runNimbleMakeLib(type: Exec) {
+//    description = 'Runs CTT_LTO=false nimble make_lib inside constantine folder'
+//    workingDir 'constantine'
+//    environment 'CTT_LTO', 'false'
+//    commandLine 'nimble', 'make_lib'
+//}
+
+task compileJavaSource(type: Exec) {
+    description = 'Compiles the Java source files'
+    commandLine 'javac', 'src/main/java/org/hyperledger/besu/nativelib/constantine/LibConstantineEIP196.java'
+}
+
+task generateJNIHeader(type: Exec) {
+    description = 'Generates the JNI header file'
+    commandLine 'javac', '-h', 'constantine/constantine-jni/include', 'src/main/java/org/hyperledger/besu/nativelib/constantine/LibConstantineEIP196.java'
+}
+
+task compileNativeLibrary(type: Exec) {
+    description = 'Compiles the native library'
+    commandLine 'clang', '-I', "${System.env.JAVA_HOME}/include", '-I', "${System.env.JAVA_HOME}/include/darwin", '-shared', '-o', 'constantine-jni/lib/libconstantine.jnilib', 'constantine-jni/src/main/c/ethereum_evm_precompiles.c', '-I', 'constantine/include', '-I', 'src/main/java/org/hyperledger/besu/nativelib/constantine', '-L', 'constantine-jni/lib', '-lconstantine'
+}
+
+task runJavaProgram(type: Exec) {
+    description = 'Runs the Java program'
+    commandLine 'java', '-Djava.library.path=constantine-jni/lib', 'org.hyperledger.besu.nativelib.constantine.LibConstantineEIP196'
+}
+
+compileNativeLibrary.dependsOn generateJNIHeader
+//generateJNIHeader.dependsOn compileJavaSource
+//compileJavaSource.dependsOn runNimbleMakeLib
+runJavaProgram.dependsOn compileNativeLibrary
+
+tasks.withType(JavaCompile) {
+    options.encoding = 'UTF-8'
+}
+
+task macArmLibCopy(type: Copy) {
+    from 'constantine-jni/lib/libconstantine.dylib'
+    into 'build/resources/main/darwin-aarch64'
+}
+
+task macLibCopy(type: Copy) {
+    from 'constantine-jni/lib/libconstantine.dylib'
+    into 'build/resources/main/darwin-x86-64'
+}
+
+task linuxLibCopy(type: Copy) {
+    from 'constantine-jni/lib/libconstantine.so'
+    into 'build/resources/main/linux-x86-64'
+}
+
+task linuxArm64LibCopy(type: Copy) {
+    from 'constantine-jni/lib/libconstantine.so'
+    into 'build/resources/main/linux-aarch64'
+}
+
```

Comment:
And you probably need a .dll for Windows, if it's one of the platforms you support.

Reply:
@garyschulte We need for windows support too?

Reply:
windows builds of besu-native is on our radar,  but we do not currently build other besu-native libs for windows.  Having Constantine build for windows would be a nice-to-have for sure.

Reply:
Okay, I will skip it for now for windows, until we do the performance comparison tests